autoload/go/jobcontrol.vim

```diff
@@ -93,7 +93,7 @@ endfunction
 " it'll be closed.
 function! s:on_exit(job_id, data)
   let std_combined = self.stderr + self.stdout
-  if empty(std_combined)
+  if jobwait([a:job_id])[0] > 0
```

Comment:
This should make it equivalent to checking v:shell_error. But yes, if there are commands that specifically avoid checking v:shell_error, then using jobcontrol would yield unexpected FAILURE. Are there any? Only one I can think of off the top of my head is the linter, which refuses to use proper return codes in a silly philosophical protest.